src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -391,6 +391,7 @@ func RenderResultsJSON(
 func renderResultsInstantaneousJSON(
 	w io.Writer,
 	result ReadResult,
+	keepNaNs bool,
```

Comment:
You'll also need to filter out NaN series entirely; you can use this:

```golang 
	// NB: if dropping NaNs, drop series with only NaNs from output entirely.
	if !opts.KeepNaNs {
		series = filterNaNSeries(series, time.Time{}, time.Now())
	}
```

Reply:
Actually, this is not the case for instant query results: here, each series contains just a single datapoint, and I check that datapoint within the series loop, effectively dropping the entire series in case of a `NaN`.